src/background/price_updates.js

```diff
@@ -85,11 +86,25 @@ async function fetchLatestPrice(product, delay) {
  * price history with the latest price.
  * @param {ExtractedProduct} data
  */
-export function updateProductWithExtracted(data) {
+export async function updateProductWithExtracted(data) {
   const state = store.getState();
   const id = getProductIdFromExtracted(data);
   const product = getProduct(state, id);
   if (product) {
-    store.dispatch(addPriceFromExtracted(data));
+    const isPriceChange = await store.dispatch(addPriceFromExtracted(data));
```

Comment:
This is only run when there's an existing product, which means we already have at least one price logged. This method has an `if` check inside that checks whether the price being added is the first price logged, but given that we only care about the return value here, and we already know that the if statement will be true, we can skip it completely.

Instead, what if `addPriceFromExtracted` returns either the price that was added or `null`? That way, we don't need to fetch the latest price below at all. If the only prices we need to fetch are the previous price and original price, we can fetch those using the previous state object and avoid calling `getState` again.